public/app/features/alerting/unified/components/rule-editor/QueryRows.tsx

```diff
@@ -110,19 +111,17 @@ export class QueryRows extends PureComponent<Props, State> {
 
       const previous = getDataSourceSrv().getInstanceSettings(item.datasourceUid);
 
+      // compatible datasource
+      if (previous?.type === settings.type) {
+        return copyModel(item, settings.uid);
+      }
+
+      // this does not make any sense, is this for backwards compat?
       if (previous?.type === settings.uid) {
-        return {
-          ...item,
-          datasourceUid: settings.uid,
-        };
+        return copyModel(item, settings.uid);
```

Comment:
The two `if` expressions above do nothing, don't they?

Reply:
Good point, after refactoring everything it's completely pointless – will remove.